src/groups/mqb/mqbblp/mqbblp_queueengineutil.cpp

```diff
@@ -801,13 +798,22 @@ void QueueEngineUtil_AppsDeliveryContext::deliverMessage()
         }
 
         if (bmqp::QueueId::k_PRIMARY_QUEUE_ID == d_queue_p->id()) {
-            // Report 'queue time' metric for all active appIds
+            const bsls::Types::Int64 timeDelta = getMessageQueueTime(
+                attributes);
+
+            // First report 'queue time' metric for entire queue
+            d_queue_p->stats()->onEvent(
+                mqbstat::QueueStatsDomain::EventType::e_QUEUE_TIME,
+                timeDelta);
+
+            // Then report 'queue time' metric for all active appIds
             bsl::vector<bslstl::StringRef>::const_iterator it =
                 d_activeAppIds.begin();
             for (; it != d_activeAppIds.cend(); ++it) {
-                QueueEngineUtil::reportQueueTimeMetric(d_queue_p->stats(),
```

Comment:
Note that with the previous implementation, with N appIds we logged queue time 2*N times:
- N times we logged duplicating metric for the entire queue
- 1*N times we logged this metric per appId

My change will remove duplications here, so I log the metric N + 1 times:
- 1 time for the entire queue
- 1*N times per appId